### PR TITLE
Split oVirt provider 

### PR DESCRIPTION
**Pull request by nasark**
- In order to facilitate the RHV provider which will be a subclass of the oVirt provider, the class names and references of `Redhat` were switched to `Ovirt`
- the `Redhat` class will now be a subclass to `Ovirt`, which will be used in the RHV provider

Depends on:
- [x] Approval of POC for RHV provider: https://github.com/nasark/manageiq-providers-red_hat_virtualization
- [x] Transfer of RHV provider: https://github.com/nasark/manageiq-providers-red_hat_virtualization
- [x] https://github.com/ManageIQ/manageiq/pull/21694
- [x] https://github.com/ManageIQ/manageiq/pull/21726
- [ ] https://github.com/ManageIQ/manageiq/pull/21698
- [ ] https://github.com/ManageIQ/manageiq-ui-classic/pull/8116
- [ ] https://github.com/ManageIQ/manageiq-decorators/pull/76
- [ ] https://github.com/ManageIQ/manageiq-providers-red_hat_virtualization/pull/6

Reference:
- https://github.com/ManageIQ/manageiq/issues/21001

### CODE REVIEW

**Comment by Fryguy** [comment]
Wow!  Very cool @nasark

File: app/models/manageiq/providers/ovirt/infra_manager.rb [hunk 1]
```diff
@@ -102,11 +102,11 @@ def populate_network_manager_connectivity(auth_url)
   end
 
   def self.ems_type
-    @ems_type ||= "rhevm".freeze
+    @ems_type ||= "ovirt".freeze
   end
 
   def self.description
-    @description ||= "Red Hat Virtualization".freeze
+    @description ||= "Ovirt".freeze
```
Comment (Fryguy) [inline app/models/manageiq/providers/ovirt/infra_manager.rb:109]:
```suggestion
    @description ||= "oVirt".freeze
```

**Comment by nasark** [comment]
Ready for review, the codeclimate and Rspec warnings originate from existing code. Let me know if there are any in particular you guys would like to see addressed.

@miq-bot add_labels enhancement
@miq-bot assign @agrare

**Comment by agrare** [comment]
This is going to take a bit of work but I'd like to get this to be green without adding the redhat provider to the Gemfile yet, this will make sure that we aren't accidentally using the old class names and also some of these failures are areas that need refactoring anyway (https://github.com/ManageIQ/manageiq/blob/master/spec/models/miq_provision_workflow_spec.rb#L133 and https://github.com/ManageIQ/manageiq/blob/master/spec/support/examples_group/shared_examples_for_provisioning_state_machine.rb#L49 for example)

File: app/models/manageiq/providers/ovirt/infra_manager.rb [hunk 2]
```diff
@@ -352,7 +352,7 @@ def self.provision_class(via)
   end
 
   def self.catalog_types
-    {"redhat" => N_("Red Hat Virtualization")}
+    {"ovirt" => N_("Ovirt")}
```
[the remark this answers is not in the document]
Comment (chessbyte) [reply app/models/manageiq/providers/ovirt/infra_manager.rb]: https://ovirt.org/

**Comment by agrare** [comment]
@miq-bot  cross-repo-tests manageiq, manageiq-ui-classic, manageiq-api

**Comment by Fryguy** [comment]
> This is going to take a bit of work but I'd like to get this to be green without adding the redhat provider to the Gemfile yet

Won't this just break devs though?  Like if they already have ManageIQ::Providers::Redhat::whatever in the database, they will be broken until that Redhat provider is reintroduced.  Or is the plan to carry _both_ providers inside this repo as a temporary thing, then extract afterwards?

**Comment by Fryguy** [comment]
Ah now I understand...sounds like a plan.

**Comment by chessbyte** [comment]
@Fryguy @agrare let's agree on the name of the Red Hat repo, especially since they have products in several of the providers we interact with -- RHV, OpenStack, OpenShift and Satellite

**Comment by agrare** [comment]
Yeah that's a good point @chessbyte the existing class names are `ManageIQ::Providers::Redhat` and changing that with a data migration is out of scope of this, but we can name the new repo to match what we would change it to.  Assume `Rhv` and `manageiq-providers-rvh` would be the choice?

**Comment by chessbyte** [comment]
@agrare up to you -- perhaps the wordier `manageiq-providers-red_hat_virtualization` would be clearer and match the various repos starting with `manageiq-providers-ibm`.  I am flexible on the name, just want to ensure we think about it.

**Comment by agrare** [comment]
@miq-bot cross-repo-tests https://github.com/ManageIQ/manageiq/pull/21698, manageiq-ui-classic, manageiq-api

**Comment by agrare** [comment]
:+1: `manageiq-providers-red_hat_virtualization` is a good name

**Comment by nasark** [comment]
Green cross repo test using PRs and `manageiq-providers-red_hat_virtualization` as overrides https://github.com/ManageIQ/manageiq-cross_repo-tests/pull/596

File: app/models/manageiq/providers/ovirt/infra_manager/event_parser.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-module ManageIQ::Providers::Redhat::InfraManager::EventParser
+module ManageIQ::Providers::Ovirt::InfraManager::EventParser
   # Sample RHV Event
```
Comment (agrare) [inline app/models/manageiq/providers/ovirt/infra_manager/event_parser.rb:2]: Minor, should probably say `# Sample Ovirt Event`

**Comment by agrare** [comment]
Hey @nasark just went through this and the other dependent PRs, except for one minor comment on the UI PR these look good.  There are a couple references to `RHV` here that we should probably update but they're mostly in strings/comments/variables no class names.

I think the next step is to transfer your `manageiq-providers-red_hat_virtualization` provider into the org (cc @Fryguy), we'd need to add that repo into the core Gemfile in lockstep with merging this.

**Comment by nasark** [comment]
@agrare @Fryguy The references to `RHV` found in strings/comments have been changed and cross repo tests are passing https://github.com/ManageIQ/manageiq-cross_repo-tests/pull/596. It would be nice to test with a live provider as well

**Comment by agrare** [comment]
Cross-repo tests are green https://github.com/ManageIQ/manageiq-cross_repo-tests/pull/596

**Pull request merged**